WebInstall/explainer_same_domain.md

```diff
@@ -96,8 +96,10 @@ const installApp = async () => {
 #### **Signatures of the `install` method (same-origin)**
 The same-origin part of the  Web Install API consists of the extension to the navigator interface with the install method. The install method can be used in several different ways. There is no difference in behaviour when this is called from a standalone window or a tab.
 
-1. `navigator.install(id[[, install_url], <params>])`: The method takes an id (and optional install url) and tries to install the current origin. If the content being installed has a manifest file, this `id` must match the value in the manifest file. If there is no manifest file present, this `id` will act as the app id for the installed content. This is relevant since if the installed content were to be given a manifest file and made into an application, there is a way to automatically update the app going forward. 
-The call can also receive an object with parameters that it can use to customize a same domain installation. These parameters alter how the app is installed and are defined in an object. More information about the parameters is found in the [Parameters](#parameters) subsection of this specification.
+1. `navigator.install()`: The method takes no parameters and tries to install the current document. If the content to be installed does not link to a manifest with a valid id, then installation will fail.
+2. `navigator.install(id, install_url, [<params>])`: The method takes an id and install url and tries to install the web content at `install_url`. If the content to be installed does not have a linked manifest that specifies an id, and the provided `id` parameter does not match the computed id, then installation will fail.
```

Comment:
So, to be clear, the API is now enforcing same-origin users to have a `id` defined in the manifest file, regardless of whether they're using the 0-param or the 2-param version.  I generally feel ok with that, given that a dev adding same-origin usage of `navigator.install` on a site should also be able to update the manifest file.

That being said, if we're requiring a `id` to be explicitly defined in the manifest now for same-origin, _does it make sense_ to have an `id` param at all?  Should it just be a "one-param" method that takes the `install_url` and optional install params?  It's not clear to me what the value/scenario is.  Would a dev at `/foo/` that is trying to install the app on `/bar/` care about the `id`?  Wouldn't they just care about installing "that app" that's at `/bar/` and not "that app but only if its id hasn't changed since I last wrote this code?

Reply:
Not every web app will have an `install_url`? I think having the 0 param makes sense to just install content that already is defined as an "app" with an id, and the 2 params is for when the developer wants to have a fine control of how to install content? (ie, custom install_url, matching ids and additional parameters.)

Reply:
There are actually two different `install_url`'s in play here:
1. The `install_url` that is (optionally) defined in the manifest.
2. The `install_url` that is the (required?) second parameter in the 2-param signature.

There's no requirement for the manifest to have an `install_url`. and even if it does, there's no requirement that the `install_url` used in the API matches the one that is optionally specified in the manifest. The `install_url` in the API signature simply tells the UA where to go to find the manifest. 

This mostly comes down to whether we are allowing the installation of web apps that don't have a manifest that at least specifies an app id.  If an app id is going to be required to be specified in the web manifest for the API to be able to install it, then I don't see a compelling argument for specifying the app id in the API signature at all (at least there isn't a well documented one in the explainer so far).  If we are allowing the API to install web apps that don't have an `id` specified in the manifest, then I think we should clearly document in the explainer with a table the full set of scenarios where an `id` is and isn't expected in the manifest.